File: android/app/src/debug/java/com/laneshadow/sandbox/stories/molecules/LSListRowStory.kt
package com.laneshadow.sandbox.stories.molecules

import androidx.compose.runtime.Composable
import com.laneshadow.theme.generated.LaneShadowTheme.IconName
import com.laneshadow.ui.molecules.LSListRow
import com.laneshadow.ui.molecules.LSListRowLeading
import com.laneshadow.ui.molecules.LSListRowTrailing
import com.nativesandbox.model.ComponentTier
import com.nativesandbox.model.Story

object LSListRowStory {
    val all: List<Story> = listOf(
        story(
            id = "molecules.listrow.leadingIcon",
            name = "Leading Icon",
            summary = "Row with icon-leading content.",
        ) {
            LSListRow(
                title = "Notifications",
                leading = LSListRowLeading.Icon(IconName.Menu),
            )
        },
        story(
            id = "molecules.listrow.leadingAvatar",
            name = "Leading Avatar",
            summary = "Row with avatar-leading content.",
        ) {
            LSListRow(
                title = "Jordan Kim",
                leading = LSListRowLeading.Avatar(initials = "JK"),
            )
        },
        story(
            id = "molecules.listrow.withSubtitle",
            name = "With Subtitle",
            summary = "Row with subtitle under title text.",
        ) {
            LSListRow(
                title = "Ride Alerts",
                subtitle = "Mentions and route updates",
                leading = LSListRowLeading.Icon(IconName.Route),
            )
        },
        story(
            id = "molecules.listrow.withToggle",
            name = "With Toggle",
            summary = "Row with trailing toggle indicator variant.",
        ) {
            LSListRow(
                title = "Location Sharing",
                leading = LSListRowLeading.Icon(IconName.Pin),
                trailing = LSListRowTrailing.Toggle(checked = true),
            )
        },
        story(
            id = "molecules.listrow.withChevron",
            name = "With Chevron",
            summary = "Row with default trailing chevron affordance.",
        ) {
            LSListRow(
                title = "Saved Routes",
                leading = LSListRowLeading.Icon(IconName.Map),
                trailing = LSListRowTrailing.Chevron,
            )
        },
        story(
            id = "molecules.listrow.withTrailingButton",
            name = "With Trailing Button",
            summary = "Row with trailing button-style label treatment.",
        ) {
            LSListRow(
                title = "Rider Profile",
                leading = LSListRowLeading.Avatar(initials = "RS"),
                trailing = LSListRowTrailing.Button(label = "Follow"),
            )
        },
    )
}

private fun story(
    id: String,
    name: String,
    summary: String,
    content: @Composable () -> Unit,
): Story =
    Story(
        id = id,
        tier = ComponentTier.Molecule,
        component = "LSListRow",
        name = name,
        summary = summary,
        content = { MoleculeStoryFrame(content) },
    )

File: ios/LaneShadow/Sandbox/Stories/Molecules/LSListRowStory.swift
import LaneShadowTheme
import NativeSandbox
import SwiftUI

@MainActor
enum LSListRowStory {
    static let all: [Story] = [
        Story(
            id: "molecules.listrow.leading-icon",
            tier: .molecule,
            component: "LSListRow",
            name: "Leading Icon",
            summary: "List row with icon-leading content and static trailing icon."
        ) { _ in
            LSListRow(
                leading: .icon(.pin),
                title: "Notifications",
                trailing: .icon(.sparkle)
            )
            .padding(Theme.shared.space.lg)
        },
        Story(
            id: "molecules.listrow.leading-avatar",
            tier: .molecule,
            component: "LSListRow",
            name: "Leading Avatar",
            summary: "List row with avatar-leading content and chevron trailing affordance."
        ) { _ in
            LSListRow(
                leading: .avatar(initials: "LS"),
                title: "Lane Shadow",
                trailing: .chevron
            )
            .padding(Theme.shared.space.lg)
        },
        Story(
            id: "molecules.listrow.with-subtitle",
            tier: .molecule,
            component: "LSListRow",
            name: "With Subtitle",
            summary: "List row with title and subtitle stack."
        ) { _ in
            LSListRow(
                leading: .icon(.route),
                title: "Route alerts",
                subtitle: "Ride alerts and mentions",
                trailing: .chevron
            )
            .padding(Theme.shared.space.lg)
        },
        Story(
            id: "molecules.listrow.with-toggle",
            tier: .molecule,
            component: "LSListRow",
            name: "With Toggle",
            summary: "List row with trailing toggle state indicator."
        ) { _ in
            LSListRow(
                leading: .icon(.sliders),
                title: "Scenic routing",
                subtitle: "Keep scenic weighting enabled",
                trailing: .toggle(isOn: true)
            )
            .padding(Theme.shared.space.lg)
        },
        Story(
            id: "molecules.listrow.with-chevron",
            tier: .molecule,
            component: "LSListRow",
            name: "With Chevron",
            summary: "Interactive list row using the chevron trailing pattern."
        ) { _ in
            LSListRow(
                leading: .icon(.map),
                title: "Open route details",
                trailing: .chevron,
                onTap: {}
            )
            .padding(Theme.shared.space.lg)
        },
        Story(
            id: "molecules.listrow.with-trailing-button",
            tier: .molecule,
            component: "LSListRow",
            name: "With Trailing Button",
            summary: "List row with trailing atom button action."
        ) { _ in
            LSListRow(
                leading: .avatar(initials: "JR"),
                title: "Justin Rider",
                subtitle: "Touring planner",
                trailing: .button(title: "Follow", variant: .outline, action: {})
            )
            .padding(Theme.shared.space.lg)
        },
    ]
}
